chaos_genius/views/anomaly_data_view.py

```diff
@@ -425,6 +357,46 @@ def kpi_anomaly_retraining(kpi_id):
         return jsonify({"msg": f"retraining failed for KPI: {kpi_id}, KPI id is None"})
 
 
+def _get_dimensions_values(kpi_id):
+    """Creates a dictionaryy of KPI dimension and their values.
+
+    :param kpi_id: ID of the KPI
+    :type row: integer
+    :return dimensions_vals: dictionary of {dimension:list(vals)}
+    :rtype: dictionary
+    """
+    # Get unique list of subdims and values from DB
+    query = (
+        db.session.query(
+            func.distinct(AnomalyDataOutput.series_type)
+        )
+        .filter(
+            (AnomalyDataOutput.kpi_id == kpi_id)
+            & (AnomalyDataOutput.anomaly_type == "subdim")
+        )
+    )
+
+    results = pd.read_sql(query.statement, query.session.bind)
+
+    if len(results) == 0:
+        current_app.logger.info("No Subdimension Anomaly Found")
+        return {}
+
+    dimension_values_dict = defaultdict(list)
+
+    # series_type strings are in the format "`dimension` == "value""
+    # split string into dimension and value to get [`dimension`, "value"]
+    dimension_values_list = list(
+        map(lambda dim_val: dim_val.split(" == "), results["distinct_1"].to_list())
+    )
+    for dim_val in dimension_values_list:
+        # remove extra quotation marks
+        dim_val_formatted = list(map(lambda string: string[1:-1], dim_val))
+        dimension_values_dict[dim_val_formatted[0]].append(dim_val_formatted[1])
```

Comment:
I would suggesting using unpacking instead of indexes
```suggestion
        dimension, values = map(lambda string: string[1:-1], dim_val)
        dimension_values_dict[dimension].append(values)
```

Reply:
Done.